cylc/flow/task_pool.py

```diff
@@ -288,7 +288,14 @@ def compute_runahead(self):
                         TASK_STATUS_SUCCEEDED,
```

Comment:
Is this test still relevant (other than in back-compat mode)?
Any task that has failed or succeeded will only be in the pool if it is incomplete?
Expired tasks immediately get removed from the pool?